.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,233 @@
+          }
+        }
+      REPO_OWNED_BY_CAMUNDA: ${{ github.repository_owner == 'camunda' }}
+      WORKFLOW_REF: ${{ github.workflow_ref }}
+    run: |
+      ## Preliminary checks
+
+      # Check if repo is owned by Camunda or an external fork
+      if [[ "${REPO_OWNED_BY_CAMUNDA}" != "true" ]]; then
+        echo "This repository does not belong to Camunda, and is probably an external fork."
+        echo "As a result, Vault secrets cannot be recovered, Camunda Nexus and DockerHub login will be disabled."
+      fi
+
+      # Check if DockerHub and DockerHub-readonly are both enabled
+      if [[ "${DOCKERHUB}" == "true" && "${DOCKERHUB_READONLY}" == "true" ]]; then
+        echo "Both dockerhub and dockerhub-readonly inputs are enabled. Only one can be enabled at a time."
+        exit 1
+      fi
+
+      # Infer team from workflow (.yml) file name if possible, otherwise default to camunda
+      worklow=$(basename ${WORKFLOW_REF%@*})  # remove Github ref part and extract filename
+      team=camunda
+      if [[ "$worklow" =~ ^(operate|optimize|tasklist|zeebe)-.*\.(yml|yaml)$ ]]; then
+        team=${BASH_REMATCH[1]}
+      fi
+
+      # Get Vault path to the DockerHub account to use, and check JSON validity at the same time
+      dockerhub_vault_path=$(echo ${DOCKERHUB_VAULT_PATHS} | jq -r --arg team "$team" '.[$team].path')
+
+      echo dockerhub-vault-path="${dockerhub_vault_path}" | tee -a $GITHUB_OUTPUT
+      echo repo-owned-by-camunda="${REPO_OWNED_BY_CAMUNDA}" | tee -a $GITHUB_OUTPUT
+    shell: bash
+  - name: Import Secrets
+    # Secrets are imported only if repo is owned by Camunda (not for external forks)
+    if: steps.checks.outputs.repo-owned-by-camunda == 'true'
+    id: secrets
+    uses: hashicorp/vault-action@v3.0.0
+    with:
+      url: ${{ inputs.vault-address }}
+      method: approle
+      roleId: ${{ inputs.vault-role-id }}
+      secretId: ${{ inputs.vault-secret-id }}
+      secrets: |
+        secret/data/github.com/organizations/camunda NEXUS_PSW | camunda-nexus-password;
+        secret/data/github.com/organizations/camunda NEXUS_USR | camunda-nexus-username;
+        ${{ steps.checks.outputs.dockerhub-vault-path }} REGISTRY_HUB_DOCKER_COM_PSW${{ inputs.dockerhub-readonly == 'true' && '_READ_ONLY' || ''}} | dockerhub-token;
+        ${{ steps.checks.outputs.dockerhub-vault-path }} REGISTRY_HUB_DOCKER_COM_USR | dockerhub-username;
+  - name: Setup Java
+    uses: actions/setup-java@v4
+    with:
+      distribution: ${{ inputs.java-distribution }}
+      java-version: ${{ inputs.java-version }}
+  - name: Setup Maven
+    if: inputs.maven-version != ''
+    uses: ./.github/actions/setup-maven-dist
+    with:
+      maven-version: ${{ inputs.maven-version }}
+      set-mvnw: true
+  # - name: Configure Maven cache
+  #   uses: ./.github/actions/setup-maven-cache
+  #   with:
+  #     maven-cache-key-modifier: ${{ inputs.maven-cache-key-modifier }}
```

Comment:
❓ Is this commented intentionally?

Reply:
Introduced by  a temporary [commit](https://github.com/camunda/camunda/pull/23785/commits/49f75af51f03abc379840b9aa68093b293e2e967) to test with cache disable, and will be removed.